definitions.md

```diff
@@ -17,74 +19,70 @@ wants to send to the [end user Application](#end-user-application).
+The Push Gateway is used for conversion and/or proxying of messages from [Application Server](#application-server) to the [Push Server](#push-server).
+If the [Application Push Protocol](#application-push-protocol) and the [Push Server Receiving Protocol](#push-server-receiving-protocol) are the same, and the [Application Server](#application-server) can reach the [Push Server](#push-server), then the gateway is not necessary.
 
-## Provider Receiving Protocol
+If the application server can not reach the push server the Push Gateway can also act as a normal Proxy, even if the [application push protocol](#application-push-protocol) and the [Push Server Receiving Protocol](#push-server-receiving-protocol) are the same.
 
-This is the protocol the [Push Provider](#push-provider) uses to receive [push messages](#push-message).
+## Push Server Receiving Protocol
 
-## Provider Push Protocol
-
-This is the protocol the [Push Provider](#push-provider) uses to send [push messages](#push-message) to the [Distributor Application](#distributor-application).
+This is the protocol the [Push Server](#push-server) uses to receive [push messages](#push-message).
 
 ## Distributor Receiving Protocol
 
-This is the protocol the [Distributor Application](#distributor-application) 
-uses to recive [push messages](#push-message) from the [Push Provider](#push-provider). 
-So it is the same as the [Provider Push Protocol](#provider-push-protocol), 
-except if the [Distributor Application](#distributor-application) 
-acts as the [Push Provider](#push-provider), 
-then there is no [Provider Push Protocol](#provider-push-protocol) 
-and this is just the [Provider Receiving Protocol](#provider-receiving-protocol).
+This is the protocol the [Push Distributor](#push-distributor)
+uses to receive [push messages](#push-message) from the [Push Server](#push-server).
+If the [Push Distributor](#push-distributor)
+acts as the [Push Server](#push-server),
+then this is just the [Push Server Receiving Protocol](#push-server-receiving-protocol).
 
 ## Rewrite Proxy
 
-If the Provider Receiving Protocol needs anything else than the URI and a GET parameter to identify the end user application (eg. header, POST parameter) or need a special structure for POST data, then a rewrite proxy is used to convert the identifier in a URI or in a GET parameter and to forge the POST parameter content structure.
+If the Push Server Receiving Protocol needs anything else than the URI and a GET parameter to identify the end user application (eg. header, POST parameter) or need a special structure for POST data, then a rewrite proxy is used to convert the identifier in a URI or in a GET parameter and to forge the POST parameter content structure.
+
+The rewrite proxy is application independent and push server dependant.
 
-The rewrite proxy is application independant and provider dependant.
+## Push Server
 
-## Push Provider
-Or Provider
+Previously: Push Provider or Provider
```

Comment:
```suggestion
Or Push Provider
Or Provider
```

That way, it is still correct to use *(push) provider* but we prefer *push server*